tests/io_components/test_metapartition.py

```diff
@@ -1645,3 +1645,32 @@ def test_parse_input_schema_formats():
     for obj in formats_obj:
         mp = parse_input_to_metapartition(obj=obj, metadata_version=4)
         assert mp.data == {"table1": df}
+
+
+def test_get_parquet_metadata(store):
+    df = pd.DataFrame({"P": np.arange(0, 10), "L": np.arange(0, 10)})
+    mp = MetaPartition(label="test_label", data={"core": df},)
+    meta_partition = mp.store_dataframes(store=store, dataset_uuid="dataset_uuid",)
+    pq_metadata = meta_partition.get_parquet_metadata(store=store, table_name="core")
+    assert pq_metadata["number_rows"] == 10
+    assert pq_metadata["number_row_groups"] == 1
+
+
+def test_get_parquet_metadata_empty_df(store):
+    df = pd.DataFrame()
+    mp = MetaPartition(label="test_label", data={"core": df},)
+    meta_partition = mp.store_dataframes(store=store, dataset_uuid="dataset_uuid",)
+    pq_metadata = meta_partition.get_parquet_metadata(store=store, table_name="core")
+    assert pq_metadata["number_rows"] == 0
+    assert pq_metadata["number_row_groups"] == 1
+
+
+def test_get_parquet_metadata_empty_metapartition(store):
+    # TODO not sure if this makes sense
+    mp = MetaPartition(label="test_label")
+    meta_partition = mp.store_dataframes(store=store, dataset_uuid="dataset_uuid",)
+    pq_metadata = meta_partition.get_parquet_metadata(store=store, table_name="core")
```

Comment:
Assertion missing here.

I am wondering: what is the difference between 
```
mp = MetaPartition(label="some_label", data={"core": pd.DataFrame()},) # empty df
```
and 
```
mp = MetaPartition(label="test_label") # no data at all
```
? And can it happen that `dispatch_metapartitions` return a MetaPartition with no data at all (not empty df)?

Reply:
the dispatch should return MPs without data. That's translates into "data was not yet loaded". If there is a dict but it is empty it means "data is loaded but it is empty"